.github/workflows/deploy.yml

```
@@ -0,0 +1,37 @@
+        run: docker build -t fastapi-app .
+
+      - name: Deploy to server
+        uses: appleboy/ssh-action@master
+        with:
+          host: 34.228.54.56  # Your EC2 instance IP
+          username: ubuntu    # Your EC2 instance username
+          key: secrets.SSH_PRIVATE_KEY  # Your SSH private key
+          script: |
+            cd /home/ubuntu/fastapi-book-project  # Replace with your project path
+            git pull origin main
+            docker-compose up -d --build
```

YAMLlint (1.35.1)

[error] no new line character at the end of file

(new-line-at-end-of-file)